docs/resources/ui-building-blocks/components/user-defined-components/component-lifecycle.md

```diff
@@ -0,0 +1,256 @@
+indicate whether a product has been favorited by the user.
+
+Follow the steps as below:
+
+<div style={{
+    position: 'relative',
+    paddingBottom: 'calc(56.67989417989418% + 41px)', // Keeps the aspect ratio and additional padding
+    height: 0,
+    width: '100%'
+}}>
+    <iframe 
+        src="https://demo.arcade.software/Y96decdgYWVll3SP9Jk8?embed&show_copy_link=true"
+        title="Get Component State"
+        style={{
+            position: 'absolute',
+            top: 0,
+            left: 0,
+            width: '100%',
+            height: '100%',
+            colorScheme: 'light'
+        }}
+        frameborder="0"
+        loading="lazy"
+        webkitAllowFullScreen
+        mozAllowFullScreen
+        allowFullScreen
+        allow="clipboard-write">
+    </iframe>
+</div>
+
+### Update Component State [Action]
+
+**Component state** values can only be updated via actions. Whenever you want to update the
+component state, call an action called **Update Component State** from the Action Flow Editor
+of the component.
+
+In the following demo, we open the Action Flow Editor on the parent widget` Conditional Builder` and
+call the **Update Component State** action to toggle the value of `isFavourite`.
+
+<div style={{
+    position: 'relative',
+    paddingBottom: 'calc(56.67989417989418% + 41px)', // Keeps the aspect ratio and additional padding
+    height: 0,
+    width: '100%'
+}}>
+    <iframe 
+        src="https://demo.arcade.software/Y96decdgYWVll3SP9Jk8?embed&show_copy_link=true"
+        title="Get Component State"
+        style={{
+            position: 'absolute',
+            top: 0,
+            left: 0,
+            width: '100%',
+            height: '100%',
+            colorScheme: 'light'
+        }}
+        frameborder="0"
+        loading="lazy"
+        webkitAllowFullScreen
+        mozAllowFullScreen
+        allowFullScreen
+        allow="clipboard-write">
+    </iframe>
+</div>
+
+#### Rebuild on Update
+
+When updating your component state in Flutter, you'll often come across the **Update
+Type** property in your Action properties. Here's what it means:
+
+- **Rebuild Containing Page:** This option triggers a re-rendering of the page
+  containing this component.
+
+- **Rebuild Current Component:** This option triggers a re-rendering of the current component only.
+
+- **No Rebuild:** Choose this option when you need to update the state value without
+  immediately reflecting the changes in the UI.
+
+:::warning[Expensive Rebuilds]
+Too many rebuilds can impact performance because rebuilding the widget tree
+frequently consumes resources and may lead to decreased responsiveness and
+increased battery usage. Therefore, it's essential to consider the trade-offs
+and use rebuilds judiciously to maintain optimal app performance.
+
+To learn more about what happens behind the scenes, refer to
+the [Generated Page] section.
```

Comment:
Generated Page in bold and add a link placeholder(#) so later we can target empty links easily.